metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageClient.scala

```diff
@@ -1,19 +1,30 @@
 package scala.meta.internal.metals
 
 import java.util.concurrent.CompletableFuture
-
 import javax.annotation.Nullable
 import org.eclipse.lsp4j.ExecuteCommandParams
 import org.eclipse.lsp4j.MessageParams
 import org.eclipse.lsp4j.MessageType
 import org.eclipse.lsp4j.jsonrpc.services.JsonNotification
 import org.eclipse.lsp4j.jsonrpc.services.JsonRequest
 import org.eclipse.lsp4j.services.LanguageClient
-
 import scala.meta.internal.tvp._
 import scala.meta.internal.metals.MetalsEnrichments._
 
 trait MetalsLanguageClient extends LanguageClient with TreeViewClient {
+  @volatile private var debuggingSupported = true
```

Comment:
Any logic related to configuring what gets forwarded to the language client should be handled in `ConfigureLanguageClient`.

As a general rule of thumb, I would recommend avoiding `val` and `var` inside traits.